v2/pkg/additional/new.go

```diff
@@ -0,0 +1,23 @@
+package additional
+
+import (
+	"github.com/openshift/oc-mirror/v2/pkg/api/v1alpha2"
+	clog "github.com/openshift/oc-mirror/v2/pkg/log"
+	"github.com/openshift/oc-mirror/v2/pkg/mirror"
+	"github.com/openshift/oc-mirror/v2/pkg/manifest"
+)
+
+
+func New(log clog.PluggableLoggerInterface,
+		config v1alpha2.ImageSetConfiguration,
+		opts mirror.CopyOptions,
+		mirror mirror.MirrorInterface,
+		manifest manifest.ManifestInterface,
+		localStorageFQDN string,
+) CollectorInterface {
+	if localStorageFQDN != "" {
+		return &LocalStorageCollector{Log: log, Config: config, Opts: opts, Mirror: mirror, Manifest: manifest, LocalStorageFQDN: localStorageFQDN}
+	} else {
+		return &Collector{Log: log, Config: config, Opts: opts, Mirror: mirror, Manifest: manifest}
+	}
```

Comment:
Nice! 
Just one thing I thought to point out is that this means we need to correctly implement, test, maintain both implementations.
For the moment, I'm not sure we have done that.

Reply:
Probably we are not going to maintain both implementations. But I agree with you that we need to create tests for the implementation we are going to maintain.